mfr/extensions/tabular/libs/stdlib_tools.py

```diff
@@ -30,3 +34,21 @@ def csv_stdlib(fp):
         })
     rows = [row for row in reader]
     return columns, rows
+
+
+def detect_quoted_fields(dialect, data):
```

Comment:
Also, the regexes should be compiled for performance, using `re.compile`.

But again, it would be more ideal if we didn't have to do regex-matching at all.